From: Jeff Mahoney <jeffm@suse.com>
Subject: [PATCH] reiserfs: fix setting of REISERFS_ATTRS on mount

 This patch fixes a problem where the REISERFS_ATTRS bit doesn't get set,
 but rather sets the value of the enum.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>

--- linux-2.6.15-rc5-mm3-clean/fs/reiserfs/super.c	2005-12-21 23:57:54.000000000 +0300
+++ linux-2.6.15-rc5-mm3/fs/reiserfs/super.c	2006-01-18 21:28:25.206460792 +0300
@@ -1131,7 +1131,7 @@
 			REISERFS_SB(s)->s_mount_opt &= ~(1 << REISERFS_ATTRS);
 		}
 	} else if (le32_to_cpu(rs->s_flags) & reiserfs_attrs_cleared) {
-		REISERFS_SB(s)->s_mount_opt |= REISERFS_ATTRS;
+		REISERFS_SB(s)->s_mount_opt |= (1 << REISERFS_ATTRS);
 	}
 }
 
